packages/itwinui-css/src/bridge.scss

```diff
@@ -252,28 +284,101 @@
     // StrataKit input background colors are different from global values
     --_iui-field-color-background: var(--ids-color-bg-control-textbox);
     --_iui-field-color-background-hover: var(--_iui-field-color-background);
-    --_iui-field-color-background-disabled: var(--ids-color-bg-control-textbox);
+    --_iui-field-color-background-disabled: var(--_iui-field-color-background);
+  }
+
+  .iui-input-flex-container {
+    // StrataKit input background colors are different from global values
+    background-color: var(--ids-color-bg-control-textbox);
+  }
+
+  .iui-input-flex-container[data-iui-disabled='true'] {
+    // StrataKit disabled inputs don't have a different background color
+    --_iui-input-background-color: var(--iui-color-background);
+
+    // iTwinUI disabled icons don't change color, but StrataKit does. Thus, handling disabled case manually.
+    svg {
```

Comment:
This will also affect nested SVGs that are inside e.g. buttons. If it looks ok during the hover state without this override, then we should keep it that way for simplicity.